### PR TITLE
File to Image conversion function

### LINKED ISSUE #5511: 5. Convert FileFields to WagtailImage objects.
> Write a function to convert a FileField (that's using an image) to a WagtailImage and attach it to a page field.

Criteria: 
- [ ] We need a conversion function that takes a FileField, checks if the file is an image, and convert that image to a WagtailImage object.
- [ ] The second parameter should be the field on the new Wagtail Page that accepts the new WagtailImage and saves the Page

### PR DESCRIPTION
**Pull request by KalobTaulien**
Closes #5511
Related PRs/issues #5489 

Criteria: 
- [x] We need a conversion function that takes a FileField, checks if the file is an image, and convert that image to a WagtailImage object.
- [x] The second parameter should be the field on the new Wagtail Page that accepts the new WagtailImage and saves the Page

---

This utility function is going to be used in the larger migration script down the road.

### CODE REVIEW

File: network-api/networkapi/buyersguide/utils.py [hunk 1]
```diff
@@ -41,3 +52,38 @@ def database_backwards(self, app_label, schema_editor, from_state, to_state):
...
+    """
+    # 1. Get the mimetype of the image.
+    mime = MimeTypes()
+    mime_type = mime.guess_type(pni_product.image.file.name)  # -> ('image/jpeg', None)
+    mime_type = mime_type[0].split('/')[1].upper()
+
+    # 2. Create an image out of the FileField.
+    pil_image = PILImage.open(pni_product.image.file)
+    f = BytesIO()
+    pil_image.save(f, mime_type)
```
Comment (Pomax) [inline network-api/networkapi/buyersguide/utils.py:72]: if we're not using `f` anywhere ese anyway:
```suggestion
    pil_image.save(BytesIO() , mime_type)
```

File: network-api/networkapi/buyersguide/utils.py [hunk 2]
```diff
@@ -41,3 +52,38 @@ def database_backwards(self, app_label, schema_editor, from_state, to_state):
...
+    """
+    Take an existing product and convert it's FileField image to a WagtailImage object.
+
+    pni_product must be an instance of a Product
+    new_product_page must be an instance of a ProductPage
+    """
+    # 1. Get the mimetype of the image.
+    mime = MimeTypes()
+    mime_type = mime.guess_type(pni_product.image.file.name)  # -> ('image/jpeg', None)
+    mime_type = mime_type[0].split('/')[1].upper()
```
Comment (Pomax) [inline network-api/networkapi/buyersguide/utils.py:67]: warning: this assumes `mime_type[0]` is splittable on `/`, even though `guess_type()` may return `None`

**Review by Pomax — CHANGES_REQUESTED** [review]
small things, but overall: nice!

File: network-api/networkapi/buyersguide/utils.py (hunk 1 above)
Comment (KalobTaulien) [reply network-api/networkapi/buyersguide/utils.py]: I was too hasty to resolve this. We do actually use `f` 

```python
    wagtail_image = WagtailImage.objects.create(
        title=new_image_name,
        file=ImageFile(f, name=new_image_name)
    )
``` 

I'm going to put this back in the next commit.

File: network-api/networkapi/buyersguide/utils.py (hunk 2 above)
Comment (KalobTaulien) [reply network-api/networkapi/buyersguide/utils.py:67]: Oh good catch. Ill add in JPEG as a default mime type if one isnt found. I'm not convinced a default JPEG mimetype is the right move here. I'm open to new ideas.

File: network-api/networkapi/buyersguide/utils.py [hunk 3]
```diff
@@ -41,3 +52,42 @@ def database_backwards(self, app_label, schema_editor, from_state, to_state):
...
+    new_product_page must be an instance of a ProductPage
+    """
+    # 1. Get the mimetype of the image.
+    mime = MimeTypes()
+    mime_type = mime.guess_type(pni_product.image.file.name)  # -> ('image/jpeg', None)
+    if mime_type:
+        mime_type = mime_type[0].split('/')[1].upper()
+    else:
+        # Default to a JPEG mimetype.
+        mime_type = 'JPEG'
```
Comment (Pomax) [inline network-api/networkapi/buyersguide/utils.py:71]: this is probably fine but we'll need to test this once we have all the code in place

**Review by Pomax — APPROVED** [review]

File: network-api/networkapi/buyersguide/utils.py (hunk 3 above)
Comment (KalobTaulien) [reply network-api/networkapi/buyersguide/utils.py:71]: I agree. Ill be doing extensive testing when we start migrating snippets to Pages (locally), too.

**Pull request merged**